service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -230,28 +246,25 @@ func equals(current, desired *v1alpha1.AppCatalogEntry) bool {
 	if current.Name != desired.Name {
 		return false
 	}
-	if !reflect.DeepEqual(current.Labels, desired.Labels) {
+	if !cmp.Equal(current.Labels, desired.Labels) {
 		return false
 	}
-	if !reflect.DeepEqual(current.Spec.Restrictions, desired.Spec.Restrictions) {
+
+	if !cmp.Equal(current.Annotations, desired.Annotations) {
 		return false
 	}
 
 	// Using reflect.DeepEqual doesn't work for the 2 date fields due to time
 	// zones. Instead we compare the unix epoch and clear the date fields.
-	if current.Spec.DateCreated.Unix() != desired.Spec.DateCreated.Unix() {
-		return false
-	}
-	current.Spec.DateCreated = nil
-	desired.Spec.DateCreated = nil
+	timeComparer := cmp.Comparer(func(current, desired *metav1.Time) bool {
```

Comment:
Using `Comparer` is more elegant than reflect since you can compare by the type matching.

Reply:
Nice TIL about Comparer. But the comment also needs updating now we don't use the unix epoch.

Reply:
I think we are still using unix epoch like before. right? 
In line number 261,
```
return current.Unix() == desired.Unix()
```

Reply:
Long week. 😂 Thanks for clarifying.